internal/hcsoci/hcsdoc_wcow.go

```diff
@@ -290,5 +285,39 @@ func createWindowsContainerDocument(ctx context.Context, coi *createOptionsInter
 	}
 	v1.MappedPipes = mpsv1
 	v2Container.MappedPipes = mpsv2
+
+	// add assigned devices to the container definition
+	parseAssignedDevices(coi, v2Container)
 	return v1, v2Container, nil
 }
+
+func specHasAssignedDevices(coi *createOptionsInternal) bool {
+	if coi.Spec.Windows != nil && coi.Spec.Windows.Devices != nil &&
+		len(coi.Spec.Windows.Devices) > 0 {
+		return true
+	}
+	return false
+}
+
+// parseAssignedDevices parses assigned devices for the container definition
+// this is currently supported for v2 argon only
+func parseAssignedDevices(coi *createOptionsInternal, v2 *hcsschema.Container) {
+	if !specHasAssignedDevices(coi) || !coi.isV2Argon() {
+		return
+	}
+
+	v2AssignedDevices := []hcsschema.Device{}
+	for _, d := range coi.Spec.Windows.Devices {
+		v2Dev := hcsschema.Device{}
+		switch d.IDType {
+		case uvm.VPCILocationPathIDType:
+			v2Dev.LocationPath = d.ID
+		case uvm.VPCIClassGUIDType:
+			v2Dev.InterfaceClassGuid = d.ID
+		default:
+			continue
```

Comment:
I think this should be an error rather than silently ignoring the unexpected device entry.

Reply:
However, I do think it would be good to check Kusto briefly to make sure no one is currently passing weird entries for devices, since that would break if we did this.

Reply:
I think the question here is really like if a user passes through a garbage device entry, do we simply ignore it and allow container create and start to function normally, or do we error saying that it is not a valid device? I'll check the kusto logs, but I think it makes sense to explicitly tell the user that their device is not valid.